tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/ContributionTask.java

```diff
@@ -0,0 +1,586 @@
+            removeConfigFile(resourceConfigPath, CONFIG_FILES.RESOURCE, remainingFiles);
+        }
+
+        Path serializationConfigPath = metadataDirectory.resolve(CONFIG_FILES.SERIALIZATION.get());
+        SerializationConfigModel serializationConfig = objectMapper.readValue(new File(serializationConfigPath.toUri()), new TypeReference<>() {});
+        if (serializationConfig.lambdaCapturingTypes().isEmpty() && serializationConfig.types().isEmpty() && serializationConfig.proxies().isEmpty()) {
+            removeConfigFile(serializationConfigPath, CONFIG_FILES.SERIALIZATION, remainingFiles);
+        }
+
+        Path jniConfigPath = metadataDirectory.resolve(CONFIG_FILES.JNI.get());
+        List<Object> jniConfig = objectMapper.readValue(new File(jniConfigPath.toUri()), new TypeReference<>() {});
+        if (jniConfig.isEmpty()) {
+            removeConfigFile(jniConfigPath, CONFIG_FILES.JNI, remainingFiles);
+        }
+
+        Path proxyConfigPath = metadataDirectory.resolve(CONFIG_FILES.PROXY.get());
+        List<Object> proxyConfig = objectMapper.readValue(new File(proxyConfigPath.toUri()), new TypeReference<>() {});
+        if (proxyConfig.isEmpty()) {
+            removeConfigFile(proxyConfigPath, CONFIG_FILES.PROXY, remainingFiles);
+        }
+
+        Path reflectConfigPath = metadataDirectory.resolve(CONFIG_FILES.REFLECTION.get());
+        List<Object> reflectConfig = objectMapper.readValue(new File(reflectConfigPath.toUri()), new TypeReference<>() {});
+        if (reflectConfig.isEmpty()) {
+            removeConfigFile(reflectConfigPath, CONFIG_FILES.REFLECTION, remainingFiles);
+        }
+
+        Path predefinedClassesConfigPath = metadataDirectory.resolve(CONFIG_FILES.PREDEFINED_CLASSES.get());
+        List<PredefinedClassesConfigModel> predefinedClassesConfig = objectMapper.readValue(new File(predefinedClassesConfigPath.toUri()), new TypeReference<>() {});
+        if (predefinedClassesConfig.size() == 1) {
+            if (predefinedClassesConfig.get(0).type().equalsIgnoreCase("agent-extracted") && predefinedClassesConfig.get(0).classes().isEmpty()) {
+                removeConfigFile(predefinedClassesConfigPath, CONFIG_FILES.PREDEFINED_CLASSES, remainingFiles);
+            }
+        }
+
+        Path agentExtractedPredefinedClasses = metadataDirectory.resolve("agent-extracted-predefined-classes");
+        if (Files.exists(agentExtractedPredefinedClasses)) {
+            File[] extractedPredefinedClasses = new File(agentExtractedPredefinedClasses.toUri()).listFiles();
+            if (extractedPredefinedClasses == null || extractedPredefinedClasses.length == 0) {
+                InteractiveTaskUtils.printUserInfo("Removing empty: agent-extracted-predefined-classes");
+                invokeCommand("rm -r " + agentExtractedPredefinedClasses, "Cannot delete empty config file: " + agentExtractedPredefinedClasses);
```

Comment:
Won't work under Windows.

Reply:
And why not use `removeConfigFile` here, too?

Reply:
Changed to: `getFileSystemOperations().delete(deleteSpec -> deleteSpec.delete(agentExtractedPredefinedClasses))`

I didn't use `removeConfigFIle` as this is not a config file but the directory that might contain predefined classes.